app/app/Http/Controllers/PasswordRecoveryController.php

```diff
@@ -79,7 +79,9 @@ public function resetPassword(Request $request)
                 'password_confirm'     => 'required|same:new_password'
             );
             $validator = Validator::make(Input::all(), $validation_rules);
-            if ($validator->fails()) {
+            if (BaseUser::authenticate($user_email, $request->input('new_password'))) {
```

Comment:
I like this change but the password reset feature isn't what Neha was referencing in the issue.

I like this change to the password reset feature because it encourages people to update their passwords when they pretty well forgot it.  The password reset form isn't navigated to unless the user clicked a link in a password recovery email.  It would be very rare for someone to hit this case but people might have forgot it enough to use the forgot password/account recovery feature and then remembered it before submitting the password reset form.

She wants the Password Change feature as you'd find from the profile to validate this.
![image](https://user-images.githubusercontent.com/11838191/43232160-96c8afde-903d-11e8-9548-9480f0294922.png)

Please add this type of check there too.  I tested the password change to be absolutely sure and it is still letting me enter the same value in the old password and new password fields.  By the way, you should have the old password as plain text so you shouldn't need to add a call to BaseUser::authenticate in the password change check.